components/tikv_util/src/memory.rs

```diff
@@ -0,0 +1,61 @@
+// Copyright 2020 TiKV Project Authors. Licensed under Apache-2.0.
+
+use crate::collections::{HashMap, HashSet};
+use kvproto::metapb::{Region, RegionEpoch};
+use std::collections::{BTreeMap, VecDeque};
+use std::hash::Hash;
+use std::mem;
+
+pub trait HeapSize {
+    fn heap_size(&self) -> usize;
+}
+
+impl HeapSize for Region {
+    #[inline]
+    fn heap_size(&self) -> usize {
+        self.start_key.len() + self.end_key.len() + mem::size_of::<RegionEpoch>()
```

Comment:
Are you sure that `region epoch` is on heap?

Reply:
Yes, it's a protobuf struct. In rust-protobuf, every message field is allocated on heap.